Wikipedia/Code/ArticleViewController+Editing.swift

```diff
@@ -77,23 +103,128 @@ extension ArticleViewController {
     
     private func presentEditor(editorViewController: UIViewController) {
         
-        let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
-        
-        let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
-        if needsIntro {
-            let editingWelcomeViewController = EditingWelcomeViewController(theme: theme) {
+        let presentEditorAction = {
+            let navigationController = WMFComponentNavigationController(rootViewController: editorViewController, modalPresentationStyle: .overFullScreen)
+            
+            let needsIntro = !UserDefaults.standard.didShowEditingOnboarding
+            if needsIntro {
+                let editingWelcomeViewController = EditingWelcomeViewController(theme: self.theme) {
+                    
+                    self.present(navigationController, animated: true)
+                }
+                editingWelcomeViewController.apply(theme: self.theme)
+                self.present(editingWelcomeViewController, animated: true) {
+                    UserDefaults.standard.didShowEditingOnboarding = true
+                }
+                
+            } else {
                 self.present(navigationController, animated: true)
             }
-            editingWelcomeViewController.apply(theme: theme)
-            present(editingWelcomeViewController, animated: true) {
-                UserDefaults.standard.didShowEditingOnboarding = true
+        }
+        
+        if authManager.permanentUser(siteURL: articleURL) == nil {
```

Comment:
I found one more weird issue with auth state stuff. This method checks for a permanentUser against the specific wiki (in my testing Test Wiki), but I didn't have a permanent state under Test Wiki in the user cache (I logged in under EN Wiki, my primary app language). So even though the app was logged in, I saw a logged out modal.

An easy fix is to just replace this line with `if !authManager.authStateIsPermanent {`. This is a better method in that it checks for a permanent status against ALL the user cache (which includes primary app wiki).